airflow/api_connexion/endpoints/extra_link_endpoint.py

```diff
@@ -18,9 +18,41 @@
 # TODO(mik-laj): We have to implement it.
 #     Do you want to help? Please look at: https://github.com/apache/airflow/issues/8140
 
+from flask import current_app
 
-def get_extra_links():
+from airflow import DAG
+from airflow.api_connexion.exceptions import NotFound
+from airflow.exceptions import TaskNotFound
+from airflow.models.dagbag import DagBag
+from airflow.models.dagrun import DagRun as DR
+from airflow.utils.session import provide_session
+
+
+@provide_session
+def get_extra_links(dag_id: str, dag_run_id: str, task_id: str, session):
     """
     Get extra links for task instance
     """
-    raise NotImplementedError("Not implemented yet.")
+    dagbag: DagBag = current_app.dag_bag
+    dag: DAG = dagbag.get_dag(dag_id)
+    if not dag:
+        raise NotFound("DAG not found")
+
+    try:
+        task = dag.get_task(task_id)
+    except TaskNotFound:
+        raise NotFound("Task not found")
+
+    execution_date = (
+        session.query(DR.execution_date).filter(DR.dag_id == dag_id).filter(DR.run_id == dag_run_id).scalar()
```

Comment:
```suggestion
        session.query(DR.execution_date).filter(DR.dag_id == dag_id, DR.run_id == dag_run_id).one()
```
`.scalar()` implicitly calls `.one()` and `one` is most suitable for Rest Api that wants to raise 404 according to sqlalchemy tutorial here https://docs.sqlalchemy.org/en/13/orm/tutorial.html#returning-lists-and-scalars